src/click_outside.rs

```diff
@@ -174,23 +176,35 @@ pub fn listen_click_outside(
 ) -> Result<ClickOutsideHandle, JsValue> {
     let (_window, document) = get_window_and_document()?;
 
-    let target_element = document
-        .get_element_by_id(element_id)
-        .ok_or_else(|| JsValue::from_str(&format!("Element with id '{element_id}' not found")))?;
+    if document.get_element_by_id(element_id).is_none() {
+        return Err(JsValue::from_str(&format!(
+            "Element with id '{element_id}' not found"
+        )));
+    }
 
     let callback = callback.clone();
-    let target_element = Rc::new(target_element);
-
+    let element_id = element_id.to_string();
     let listener = EventListener::new(&document, "click", move |event| {
-        if let Some(mouse_event) = event.dyn_ref::<MouseEvent>()
-            && let Some(click_target) = mouse_event.target()
+        let Some(mouse_event) = event.dyn_ref::<MouseEvent>() else {
+            return;
+        };
+
+        let Some(window) = web_sys::window() else {
+            return;
+        };
+        let Some(document) = window.document() else {
+            return;
+        };
```

Comment:
Is this part of the code written intentionally in anticipation of changes to `window` and `document`?
I’m asking because, similar to the `let document_for_closure = document.clone();` statement used in the `add_listen_click` and `add_listen_click_complex` functions, applying the same approach here could reduce the repeated calls on each click.